File: ChatSockets/persistence/ICrudRepository.cs
using System.Collections.Generic;
using chat.model;
namespace chat.persistence
{
    public interface ICrudRepository<ID,T> where T:IIdentifiable<ID>
    {
        void save(T entity);
        void delete(ID id);
        T findOne(ID id);
        void update(ID id, T e);
        IEnumerable<T> getAll();
    }
}

File: JavaChatSockets/ChatPersistence/src/main/java/chat/persistence/ICrudRepository.java
package chat.persistence;

import chat.model.Identifiable;

/**
 * Created by grigo on 3/17/17.
 */
public interface ICrudRepository<ID, E extends Identifiable<ID>> {
    void save(E e);
    void delete(ID id);
    E findOne(ID id);
    void update(ID id, E e);
    Iterable<E> getAll();
}
